server/src-lib/Hasura/GraphQL/Validate/Field.hs

```diff
@@ -49,13 +50,15 @@ type ArgsMap = Map.HashMap G.Name AnnInpVal
 
 type SelSet = Seq.Seq Field
 
--- N.B. This is a tree via 'SelSet'
 -- | https://graphql.github.io/graphql-spec/June2018/#sec-Language.Fields 
+--
+-- N.B. This is a tree via 'SelSet'
 data Field
   = Field
   { _fAlias     :: !G.Alias
   , _fName      :: !G.Name
   , _fType      :: !G.NamedType
+  -- ^ TODO: this seems to get initialized to "unknown.."; how is this used? Document or improve.
```

Comment:
This is initialized to `unknown_type` only in the RemoteJoins module. This is because we are constructing a nested `Field` and we don't have the "named type" of the parent fields. It's not important also as `Field` is more annotated than required for constructing a remote query.

Reply:
Maybe I'll make a smart constructor for this where we can put docs